### ISSUE TITLE
Glob failing on matching `/` with Index was outside the bounds of the array

### ISSUE THREAD

**Issue by xoofx** [dazinator/DotNet.Glob#82]
Hey,

I have started to use this library and it's great, thank you!

I have discovered one small glitch. If you have the following code:

```c#
var glob = Glob.Parse("**/~*");
// throws System.IndexOutOfRangeException
glob.IsMatch("/");
```

The `glob.IsMatch` will throw a `System.IndexOutOfRangeException`

**Comment by dazinator** [comment]
Thanks for reporting- will take a look soon!

**Comment by dazinator** [comment]
@xoofx - This has been fixed in version 3.1.1, which should appear on nuget shortly (or also latest alpha).
Many thanks for reporting this, and sorry about the bug.
